sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context, Result};
+use arrayvec::ArrayVec;
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Bit {
+    Low = 0,
+    High = 1,
+}
+
+impl std::convert::From<u8> for Bit {
```

Comment:
Nit: `From` is in the prelude

```suggestion
impl From<u8> for Bit {
```

Reply:
Also, I think we should implement `TryFrom` here and return an error if the input is not `0` or `1`. Panics are only really for bugs and not misuse.

Reply:
Removed the `panic` because the in practice it is either 0 or 1.